src/float.jl

```diff
@@ -0,0 +1,6 @@
+Base.eps(T::Type{<:Union{ArfOrRef,ArbOrRef,AcbOrRef}}) = eps(one(T))
+Base.eps(x::ArfOrRef) = Arf(set_ulp!(Mag(), x, prec = precision(x)), prec = precision(x))
+Base.eps(x::ArbOrRef) =
```

Comment:
I think we could save one allocation here (at the cost of readability):
```julia
function eps(x::ArbOrRef)
    prec= precision(x)
    out = Arb(prec=prec)
    rad = radref(out)
    set_ulp!(rad, midref(x), prec=prec)
    set!(midref(out), rad)
    Arblib.zero!(rad)
    return out
end
```
with benchmarks for `x=Arb(rand())`:
```julia
BenchmarkTools.Trial: # old eps
  memory estimate:  128 bytes
  allocs estimate:  3
  --------------
  minimum time:     72.061 ns (0.00% GC)
  median time:      75.540 ns (0.00% GC)
  mean time:        301.690 ns (44.43% GC)
  maximum time:     163.718 μs (80.91% GC)
  --------------
  samples:          10000
  evals/sample:     968
```
```julia
BenchmarkTools.Trial: # new eps
  memory estimate:  96 bytes
  allocs estimate:  2
  --------------
  minimum time:     47.383 ns (0.00% GC)
  median time:      49.274 ns (0.00% GC)
  mean time:        169.549 ns (43.90% GC)
  maximum time:     134.622 μs (84.70% GC)
  --------------
  samples:          10000
  evals/sample:     984
```

as for the (mathematical) definition, I'm fine with what is here;
I tried to solicit some advice on this, but got cryptic/no answers, see
https://discourse.julialang.org/t/doublefloats-jl-double128-quad64/51049/55?u=abulak